### DISCUSSION TITLE
Swapping skin for node?

### DISCUSSION THREAD

**Discussion by MichalSob1238** [eckig/graph-editor#107] [Q&A] [answered]
How would one go about swapping what skin is used to represent a node?
The skins are associated with nodes using the type field.
Say we have a "default-node" skin and a "changed-node" skin, both properly associated with their Skin classes.
We have implemented a functionality that after double-clicking a node changes its type in the model from one to the other.
How can we inform the skin side of this change, remove the old skin and display the new one?

**Comment by eckig** [comment, marked as answer]
I looked into it: There is already a change listener on the "type" property of the `GNode`.

So, (assuming you registered the skins for each type correctly) the skins should be swapped accordingly, when the type is changed.

**Reply by MichalSob1238** [reply]
Yes, it does indeed work. My mistake was caused by the fact that this does not update the view  - another action, such as adding a node, is required for the view to be updated and the new skin to be displayed.

Manually calling getGraphEditor().reload(); afterwards solved the issue.
Thanks for the help.
